bigml/ensemble.py

```diff
@@ -67,25 +76,34 @@ def __init__(self, ensemble, api=None, max_models=None):
             self.api = BigML(storage=STORAGE)
         else:
             self.api = api
+        self.resource_id = None
+        # to be deprecated
         self.ensemble_id = None
+        self.objective_id = None
         self.distributions = None
         self.models_splits = []
         self.multi_model = None
         if isinstance(ensemble, list):
-            try:
-                models = [get_model_id(model) for model in ensemble]
-            except ValueError:
-                raise ValueError('Failed to verify the list of models. Check '
-                                 'your model id values.')
+            if all([isinstance(model, Model) for model in ensemble]):
+                models = ensemble
+                self.model_ids = [local_model.resource_id for local_model in
+                                  models]
+            else:
+                try:
+                    models = [get_model_id(model) for model in ensemble]
+                    self.model_ids = models
+                except ValueError:
+                    raise ValueError('Failed to verify the list of models. Check '
+                                     'your model id values.')
```

Comment:
perhaps you could append here the message of the original exception, to help in diagnosing what's the wrong value...